### PR TITLE
Migration of NAS tests

### PR DESCRIPTION
**Pull request by ultmaster**
### Description ###

Migrate NAS tests to full tests.

In addition, this PR also includes:

- Improvements of serializer documents and error messages.
- Support trial command templating in Retiarii experiment.
- Rewrite multi-trial test with multiprocessing.
- Add a quick test for expeirment on UT.
- Remove unused `it.sh` and `nas.sh`.
- Remove annoying debugging messages in `test_kill_command`.
- Support double-quotes in trial command for local training service.

#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [x] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### How to test ###

N/A

### CODE REVIEW

File: ts/nni_manager/training_service/local/localTrainingService.ts [hunk 1]
```diff
@@ -414,16 +414,18 @@ class LocalTrainingService implements TrainingService {
     private getScript(workingDirectory: string): string[] {
         const script: string[] = [];
         if (process.platform === 'win32') {
+            const escapedCommand = this.config.trialCommand.replace('"', '`"');
```
Comment (liuzhe-lz) [inline ts/nni_manager/training_service/local/localTrainingService.ts:417]: Please try `shellString()` in `common/shellUtils.ts`.

File: nni/retiarii/experiment/config/experiment_config.py [hunk 2]
```diff
@@ -31,6 +32,7 @@ class RetiariiExeConfig(ExperimentConfig):
     trial_code_directory: utils.PathLike = '.'
     trial_command: str = '_reserved'
     # new config field for NAS
+    trial_command_params: Optional[Dict[str, Any]] = None
```
Comment (liuzhe-lz) [inline nni/retiarii/experiment/config/experiment_config.py:35]: Can you prefix it with underscore?

File: nni/retiarii/experiment/config/experiment_config.py [hunk 3]
```diff
@@ -46,15 +51,25 @@ def _canonicalize(self, _parents):
...
+
+        _trial_command_params = {
+            # Default variables
+            'envs': '',
+            # TODO: maybe use sys.executable rendered in trial side (e.g., trial_runner)
+            'python': sys.executable,
+            'execution_engine': self.execution_engine.name,
+
+            # Overridden by new
+            **(self._trial_command_params or {})
```
Comment (QuanluZhang) [inline nni/retiarii/experiment/config/experiment_config.py:70]: according to line 73, this line does not take any effect. so why add it?

Comment (ultmaster) [reply nni/retiarii/experiment/config/experiment_config.py:70]: Why no effect?

**Review by QuanluZhang — APPROVED** [review]

**Review by J-shang — APPROVED** [review]

File: test/algo/nas/test_cgo_engine.py [hunk 4]
```diff
@@ -162,7 +162,7 @@ def _new_trainer():
 
 
 def _load_mnist(n_models: int = 1):
-    path = Path(__file__).parent / 'mnist_pytorch.json'
+    path = Path('ut/nas/mnist_pytorch.json')
```
Comment (liuzhe-lz) [inline test/algo/nas/test_cgo_engine.py:165]: I suggest to keep `__file__`. It is dangerous to assert pwd.

**Review by liuzhe-lz — APPROVED** [review]

File: test/algo/nas/test_cgo_engine.py (hunk 4 above)
Comment (ultmaster) [reply test/algo/nas/test_cgo_engine.py:165]: It should be ``pytestconfig.rootpath / ...``. But the fixture doesn't work well with python unittest module. And it's a CGO engine test -- I don't want to get my hands dirty.

**Pull request merged**